R/derive_extreme_records.R

```diff
@@ -0,0 +1,157 @@
+#' Add the First or Last Observation for Each By Group as New Records
+#'
+#' Adds the first or last observation for each by group as new observations. It
+#' can be used for example for adding the maximum or minimum value as a separate
+#' visit.
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by the `order` and the `by_vars` parameter are
+#'   expected.
+#'
+#' @param by_vars Grouping variables
+#'
+#'   *Permitted Values:* list of variables created by `vars()`
+#'
+#' @param order Sort order
+#'
+#'   Within each by group the observations are ordered by the specified order.
+#'
+#'   *Permitted Values:* list of variables or `desc(<variable>)` function calls
+#'   created by `vars()`, e.g., `vars(ADT, desc(AVAL))`
+#'
+#' @param mode Selection mode (first or last)
+#'
+#'   If `"first"` is specified, the first observation of each by group is added
+#'   to the input dataset. If `"last"` is specified, the last observation of
+#'   each by group is added to the input dataset.
+#'
+#'   *Permitted Values:* `"first"`, `"last"`
+#'
+#' @param check_type Check uniqueness?
+#'
+#'   If `"warning"` or `"error"` is specified, the specified message is issued
+#'   if the observations of the filtered input dataset are not unique with
+#'   respect to the by variables and the order.
+#'
+#'   *Default:* `"warning"`
+#'
+#'   *Permitted Values:* `"none"`, `"warning"`, `"error"`
+#'
+#' @param filter Filter for observations to consider
+#'
+#'   Only observations fulfilling the specified condition are taken into account
+#'   for selecting the first or last observation. If the parameter is not
+#'   specified, all observations are considered.
+#'
+#'   *Default*: `NULL`
+#'
+#'   *Permitted Values*: a condition
+#'
+#' @param set_values_to Variables to be set
+#'
+#'   The specified variables are set to the specified values for the new
+#'   observations. For example `vars(DTYPE = "MINIMUM")` defines the derivation
+#'   type for the new observations.
+#'
```

Comment:
In `derive_summary_records()` the def is:
#'   Set a list of variables to some specified value for the new observation(s)
#'   + LHS refer to a variable.
#'   + RHS refers to the values to set to the variable. This can be a string, a symbol, a numeric
#'   value or NA.
#'   (e.g.  `vars(PARAMCD = "TDOSE",PARCAT1 = "OVERALL")`).
#'   More general expression are not allowed.

Should it be the same?

Reply:
Yes, I have unified the description in `derive_summary_records()` and `derive_extreme_records()`.